Palace/Network/TPPNetworkExecutor.swift

```diff
@@ -78,11 +78,33 @@ extension TPPNetworkExecutor: TPPRequestExecuting {
   /// the network or from the cache.
   /// - Returns: The task issueing the given request.
   @discardableResult
-  func executeRequest(_ req: URLRequest,
-           completion: @escaping (_: NYPLResult<Data>) -> Void) -> URLSessionDataTask {
-    let task = urlSession.dataTask(with: req)
+  func executeRequest(_ req: URLRequest, completion: @escaping (_: NYPLResult<Data>) -> Void) -> URLSessionDataTask {
+    
+    if !TPPUserAccount.sharedAccount().authTokenHasExpired {
+      return performDataTask(with: req, completion: completion)
+    }
+    
+    refreshTokenAndResume(task: nil) { [weak self] newToken in
+      guard let strongSelf = self else { return }
+      
+      if let token = newToken {
+        var updatedRequest = req
+        updatedRequest.setValue("Bearer \(token)", forHTTPHeaderField: "Authorization")
+        strongSelf.executeRequest(updatedRequest, completion: completion)
+
+      } else {
+        completion(NYPLResult.failure(NSError(domain: "com.PalaceApp.error", code: 401, userInfo: [NSLocalizedDescriptionKey: "Unauthorized HTTP"]), nil))
```

Comment:
There's TPPErrorLogger with error codes, and code `401` means audiobook is corrupted ([here](https://github.com/ThePalaceProject/ios-core/blob/main/Palace/Logging/TPPErrorLogger.swift)). You may need to add a new code there.

Also, we have `TPPErrorLogger.clientDomain` for NSError domains to not type domain values everywhere, but I think it was used once or twice so far.

Reply:
Ah, thanks, was looking for this.